src/goto-instrument/contracts/contracts.cpp

```diff
@@ -38,21 +38,62 @@ Date: February 2016
 // This is used in the implementation of multidimensional decreases clauses.
 static exprt create_lexicographic_less_than(
   const std::vector<symbol_exprt> &lhs,
-  const std::vector<symbol_exprt> &rhs);
+  const std::vector<symbol_exprt> &rhs)
+{
+  PRECONDITION(lhs.size() == rhs.size());
+
+  if(lhs.empty())
+  {
+    return false_exprt();
+  }
+
+  // Store conjunctions of equalities.
+  // For example, suppose that the two input vectors are <s1, s2, s3> and <l1,
+  // l2, l3>.
+  // Then this vector stores <s1 == l1, s1 == l1 && s2 == l2,
+  // s1 == l1 && s2 == l2 && s3 == l3>.
+  // In fact, the last element is unnecessary, so we do not create it.
+  exprt::operandst equality_conjunctions(lhs.size());
+  equality_conjunctions[0] = binary_relation_exprt(lhs[0], ID_equal, rhs[0]);
+  for(unsigned int i = 1; i < equality_conjunctions.size() - 1; i++)
+  {
+    binary_relation_exprt component_i_equality{lhs[i], ID_equal, rhs[i]};
+    equality_conjunctions[i] =
+      and_exprt(equality_conjunctions[i - 1], component_i_equality);
+  }
```

Comment:
```suggestion
  exprt::operandst equality_conjunctions;
  for(unsigned int i = 0; i < lhs.size(); i++)
  {
    binary_relation_exprt component_i_equality{lhs[i], ID_equal, rhs[i]};
    equality_conjunctions.push_back(component_i_equality);
  }
  exprt equality_constraint = conjunction(equality_conjunctions);
```

Reply:
You'd also need to update the next loop in the same style. Perhaps, even combine them together? Just a suggestion, note a hard requirement.

Reply:
So the `equality_conjunctions` vector stores a list of conjunctions. In your suggestion, it only stores equality predicates and we compute a single conjunction at the end. That wouldn't work.

Basically we want to have `lhs[0] == rhs[0]`, `lhs[0] == rhs[0] && lhs[1] == rhs[1]`, and so on in the `equality_conjunctions` vector.